client/state/posts/utils/is-terms-equal.js

```diff
@@ -10,9 +8,24 @@ export function isTermsEqual( localTermEdits, savedTerms ) {
 	return Object.entries( localTermEdits ).every( ( [ taxonomy, terms ] ) => {
 		const termsArray = Object.values( terms );
 		const isHierarchical = typeof termsArray[ 0 ] === 'object' && termsArray[ 0 ] !== null;
-		const normalizedEditedTerms = isHierarchical ? map( termsArray, 'ID' ) : termsArray;
+		const normalizedEditedTerms = new Set(
+			isHierarchical ? termsArray.map( ( term ) => term.ID ) : termsArray
+		);
 		const normalizedKey = isHierarchical ? 'ID' : 'name';
-		const normalizedSavedTerms = map( savedTerms[ taxonomy ], normalizedKey );
-		return ! xor( normalizedEditedTerms, normalizedSavedTerms ).length;
+		const normalizedSavedTerms = new Set(
+			Object.values( savedTerms[ taxonomy ] ?? {} ).map( ( value ) => value[ normalizedKey ] )
```

Comment:
Nit: would be ideal if we perform the `map()` only if it's necessary - it's not necessary on an empty array.

Reply:
Added https://github.com/Automattic/wp-calypso/pull/81682/commits/1355bdfa760f9997c3231ced36a8842baa123699 which circumvents the `.map` on an empty array although my assumption would be that JS engines optimize this in a way so it doesn't really make a huge difference. I'm fine with that change though 👍🏻